src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs

```diff
@@ -14,54 +14,502 @@
+            HelpMessage = Constants.HelpRepoConfigurationLastCommitId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoVstsConfig,
+            Mandatory = false,
+            HelpMessage = Constants.HelpRepoConfigurationLastCommitId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoGitConfig,
+            Mandatory = false,
+            HelpMessage = Constants.HelpRepoConfigurationLastCommitId)]
+        [ValidateNotNullOrEmpty]
+        #endregion // Attributes
+        public string LastCommitId { get; set; }
+
+        #region Attributes
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByInputObjectFactoryRepoVstsConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationProjectName)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryNameFactoryRepoVstsConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationProjectName)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoVstsConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationProjectName)]
+        [ValidateNotNullOrEmpty]
+        #endregion // Attributes
+        public string ProjectName { get; set; }
+
+        #region Attributes
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByInputObjectFactoryRepoVstsConfig,
+            Mandatory = false,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryNameFactoryRepoVstsConfig,
+            Mandatory = false,
+            HelpMessage = Constants.HelpRepoConfigurationTenantId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoVstsConfig,
+            Mandatory = false,
+            HelpMessage = Constants.HelpRepoConfigurationTenantId)]
+        [ValidateNotNullOrEmpty]
+        #endregion // Attributes
+        public string TenantId { get; set; }
+
+        #region Attributes
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByInputObjectFactoryRepoGitConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpTagsForFactory)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByFactoryNameFactoryRepoGitConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationHostName)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceIdFactoryRepoGitConfig,
+            Mandatory = true,
+            HelpMessage = Constants.HelpRepoConfigurationHostName)]
+        [ValidateNotNullOrEmpty]
+        #endregion // Attributes
+        public string HostName { get; set; }
+
         public override void ExecuteCmdlet()
         {
+            this.ByInputObject();
```

Comment:
@sanjkuma minor: can you create a new method `ValidateParameters()` that then calls these two methods, or even moves the contents of the two methods into this single method?

Reply:
Done